server/Arcadia.Assistant.CSP/ArcadiaHealthChecker.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Arcadia.Assistant.CSP
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Akka.Actor;
+    using Configuration.Configuration;
+    using Health.Abstractions;
+
+    public class ArcadiaHealthChecker : HealthChecker
+    {
+        private const string VacationRegistryActorPath = @"/user/organization/employees/vacations-registry";
+        private const string DepartmentStorageActorPath = @"/user/organization/departments/departments-storage";
+        private const string EmployeesInfoStorageActorPath = @"/user/organization/employees/employees-storage";
+
+        private readonly string[] HealthStateNames = new[] { "Vacations Registry", "Departments Storage", "Employees Info Storage" };
+
+        private readonly AppSettings settings;
+
+        public ArcadiaHealthChecker(AppSettings settings)
+        {
+            this.settings = settings;
+        }
+
+        protected override async Task<HealthCheckMessageResponse> GetHealthStates()
+        {
+            var vacationsRegistry = Context.ActorSelection(VacationRegistryActorPath);
+            var departmentsStorage = Context.ActorSelection(DepartmentStorageActorPath);
+            var employeesInfoStorage = Context.ActorSelection(EmployeesInfoStorageActorPath);
+
+            var result = await Task.WhenAll(
+                this.GetActorHealthState(vacationsRegistry),
+                this.GetActorHealthState(departmentsStorage),
+                this.GetActorHealthState(employeesInfoStorage));
+
+            var healthStates = result
+                .Select((x, i) => new
+                {
+                    Name = HealthStateNames[i],
+                    State = x
+                })
+                .ToDictionary(x => x.Name, x => x.State);
+
+            return new HealthCheckMessageResponse(healthStates);
+        }
+
+        private async Task<HealthState> GetActorHealthState(ActorSelection actor)
+        {
+            try
+            {
+                var result = await actor.Ask<GetHealthCheckStatusMessage.GetHealthCheckStatusResponse>(
+                    GetHealthCheckStatusMessage.Instance,
+                    settings.Timeout);
```

Comment:
Ok, what I'm afraid of is that if THIS times out, the calling actor (web-part) will also time out, so we won't know which one timed out. That's why I suggested to use default timeout / 2

Reply:
ArcadiaHealthChecker actor is called with timeout * 2 value.

Reply:
https://github.com/SoftwareCountry/arcadian-assistant/pull/245/files#diff-3f2b01ecf199698605c561a0e4a62ce8R34

Reply:
Nevermind, since you have it in a separate variable we can change it